src/output.cpp

```diff
@@ -0,0 +1,736 @@
+      "                         or a particle restart file\n"
+      "  -s, --threads          Number of OpenMP threads\n"
+      "  -t, --track            Write tracks for all particles\n"
+      "  -v, --version          Show version information\n"
+      "  -h, --help             Show this message\n";
+  }
+}
+
+//==============================================================================
+
+void print_version()
+{
+  if (mpi::master) {
+    std::cout << "OpenMC version " << VERSION_MAJOR << '.' << VERSION_MINOR
+      << '.' << VERSION_RELEASE << '\n';
+#ifdef GIT_SHA1
+    std::cout << "Git SHA1: " << GIT_SHA1 << '\n';
+#endif
+    std::cout << "Copyright (c) 2011-2019 Massachusetts Institute of "
+      "Technology and OpenMC contributors\nMIT/X license at "
+      "<http://openmc.readthedocs.io/en/latest/license.html>\n";
+  }
+}
+
+//==============================================================================
+
+void print_columns()
+{
+  if (settings::entropy_on) {
+    std::cout <<
+      "  Bat./Gen.      k       Entropy         Average k \n"
+      "  =========   ========   ========   ====================\n";
+  } else {
+    std::cout <<
+      "  Bat./Gen.      k            Average k\n"
+      "  =========   ========   ====================\n";
+  }
+}
+
+//==============================================================================
+
+void print_generation()
+{
+  // Save state of cout
+  auto f {std::cout.flags()};
+
+  // Determine overall generation and number of active generations
+  int i = overall_generation() - 1;
+  int n = simulation::current_batch > settings::n_inactive ?
+    settings::gen_per_batch*simulation::n_realizations + simulation::current_gen : 0;
+
+  // Set format for values
+  std::cout << std::fixed << std::setprecision(5);
+
+  // write out information batch and option independent output
+  std::cout << "  "  << std::setw(9) <<  std::to_string(simulation::current_batch)
+    + "/" + std::to_string(simulation::current_gen) << "   " << std::setw(8)
+    << simulation::k_generation[i];
+
+  // write out entropy info
+  if (settings::entropy_on) {
+    std::cout << "   " << std::setw(8) << simulation::entropy[i];
+  }
+
+  if (n > 1) {
+    std::cout << "   " << std::setw(8) << simulation::keff << " +/-"
+      << std::setw(8) << simulation::keff_std;
+  }
+  std::cout << '\n';
+
+  // Restore state of cout
+  std::cout.flags(f);
+}
+
+//==============================================================================
+
+void print_batch_keff()
+{
+  // Save state of cout
+  auto f {std::cout.flags()};
+
+  // Determine overall generation and number of active generations
+  int i = simulation::current_batch*settings::gen_per_batch - 1;
+  int n = simulation::n_realizations*settings::gen_per_batch;
+
+  // Set format for values
+  std::cout << std::fixed << std::setprecision(5);
+
+  // write out information batch and option independent output
```

Comment:
Question: Can you clarify "option independent output"?  Did you mean "optional independent output"? Or does "option independent" refer specifically to `gen_per_batch` and the overall generation index in `k_generation `?